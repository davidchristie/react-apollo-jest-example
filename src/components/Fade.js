import React, { Component } from 'react'
import { CSSTransitionGroup } from 'react-transition-group'
import { Collapse } from 'reactstrap'

export default class Fade extends Component {
  constructor (props) {
    super(props)
    this.state = {
      open: false
    }
  }

  componentDidMount () {
    this.setState({
      open: true
    })
  }

  render () {
    return (
      <Collapse isOpen={this.state.open}>
        <CSSTransitionGroup
          transitionAppear
          transitionAppearTimeout={1000}
          transitionEnter={false}
          transitionLeaveTimeout={1000}
          transitionName='fade'
        >
          {this.props.children}
        </CSSTransitionGroup>
      </Collapse>
    )
  }
}
