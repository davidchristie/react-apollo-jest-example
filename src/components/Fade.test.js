import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Fade from './Fade'

describe('Fade component', () => {
  it('should render correctly', () => {
    expect(toJson(mount(<Fade />))).toMatchSnapshot()
  })
})
