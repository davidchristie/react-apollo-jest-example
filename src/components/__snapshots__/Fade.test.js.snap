// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fade component should render correctly 1`] = `
<Fade>
  <Collapse
    delay={
      Object {
        "hide": 350,
        "show": 350,
      }
    }
    isOpen={true}
    onClosed={[Function]}
    onOpened={[Function]}
    tag="div"
  >
    <div
      className="collapsing"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={1000}
        transitionEnter={false}
        transitionLeave={true}
        transitionLeaveTimeout={1000}
        transitionName="fade"
      >
        <TransitionGroup
          childFactory={[Function]}
          component="span"
          transitionAppear={true}
          transitionAppearTimeout={1000}
          transitionEnter={false}
          transitionLeave={true}
          transitionLeaveTimeout={1000}
          transitionName="fade"
        >
          <span />
        </TransitionGroup>
      </CSSTransitionGroup>
    </div>
  </Collapse>
</Fade>
`;
